blog/2020-09-14-debezium-auto-create-topics.adoc

```diff
@@ -0,0 +1,235 @@
+that fits the needs of these internal topics.
+
+When you start a Debezium connector the topics for the captured events are created by the Kafka
+broker based on a default, maybe customized, configuration in the broker if
+`auto.create.topics.enable = true` is enabled in the broker config:
+
+[source,options="nowrap",shell]
+----
+auto.create.topics.enable = false
+default.replication.factor = 1
+num.partitions = 1
+compression.type = producer
+log.cleanup.policy = delete
+log.retention.ms = 604800000  ## 7 days
+----
+
+But often, when you use Debezium and Kafka in a production environment you might often choose to disable
+Kafka's topic auto creation capability with `auto.create.topics.enable = false`, or you want the
+connector topics to be configured differently from the default. In this case you have to create
+topics for Debezium's captured data sources upfront.{empty} +
+But there's good news! Since Kafka Connect version 2.6.0 and above this can be now automated since
+https://cwiki.apache.org/confluence/display/KAFKA/KIP-158%3A+Kafka+Connect+should+allow+source+connectors+to+set+topic-specific+settings+for+new+topics[KIP-158]
+is implemented to enable customizable topic creation with Kafka Connect.
+
+== Kafka Connect
+
+Kafka Connect since Kafka 2.6.0 comes with topic creation enabled:
+
+[source,options="nowrap",shell]
+----
+topic.creation.enable = true
+----
+
+If you don't want to allow automatic topic creation by connectors you can set this value to `false`
+in the Kafka Connect config (_connect-distributed.properties_ file or via environment variable
+_CONNECT_TOPIC_CREATION_ENABLE_).
+
+== Update Connector Configuration
+
+Kafka Connect topic creation works with groups. There's always a `default` group which is used when
+there's no other group defined that matches the topic.
+
+Every group can specify a collection of topic configuration properties, and a regular expression list
+of topic names that config should apply to.
+
+You can specify all https://kafka.apache.org/documentation/#topicconfigs[*topic level configuration parameters*]
+to customize how the matched topics of the group will be created.
+
+Let's see how we can extend this Postgres config for the Kafka Connect topic creation:
+
+[source,options="nowrap",json]
+----
+{
+    "name": "inventory-connector",
+    "config": {
+        "connector.class": "io.debezium.connector.postgresql.PostgresConnector",
+        "tasks.max": "1",
+        "database.hostname": "postgres",
+        "database.port": "5432",
+        "database.user": "postgres",
+        "database.password": "postgres",
+        "database.dbname" : "postgres",
+        "database.server.name": "dbserver1",
+        "schema.whitelist": "inventory"
+    }
+}
+----
+
+=== Default Config
+
+All topics not matching other `topic.creation` groups will apply the `default` group
+config.{empty} +
+As default we want `replication.factor = 1`, `partitions = 5`, the topic should be key
```

Comment:
Se above, repl factor 3?

Reply:
yes, same here. stay with the example repo code or diverge in the blog post?

Reply:
Diverging from the example is fine, if it makes for a more realistic snippet in the post.

Reply:
Like said above, I will keep the absolute default values above and update these examples here.